contrib/pkg/awsprivatelink/common/vars.go

```diff
@@ -0,0 +1,11 @@
+package common
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	CredsSecret   *corev1.Secret
+	DynamicClient client.Client
+)
```

Comment:
The CredsSecret and DynamicClient variables are defined in the `common` package (instead of the `awsprivatelink` package) to avoid import cycles like awsprivatelink -> endpointvpc -> awsprivatelink.

Reply:
Ack.

Though I wonder if it would look cleaner to put these in a struct that is then composed into the `*Options` of the subcommands? It would be nice to be able to avoid referring to `common.Thing` in the meat of the code.

Reply:
Thanks for pointing that out. I guess I will keep it in mind and save this to future contributions.